cdap-ui/app/features/operations/ops-ctrl.js

```diff
@@ -178,8 +190,46 @@ angular.module(PKG.name+'.feature.dashboard')
       return {title : 'System metrics', columns : widgets};
     }
 
+    function startPolling (widget) {
+      dataSrc.poll({
+        _cdapPath: '/metrics/query',
+        method: 'POST',
+        body: MyMetricsQueryHelper.constructQuery(
+          'qid',
+          MyMetricsQueryHelper.contextToTags(widget.metric.context),
+          widget.metric
+        )
+      }, function (res) {
+        var processedData = MyChartHelpers.processData(
+          res,
+          'qid',
+          widget.metric.names,
+          widget.metric.resolution,
+          widget.settings.aggregate
+        );
+
+        processedData = MyChartHelpers.c3ifyData(processedData, widget.metric, widget.metricAlias);
+        var data = {
+          x: 'x',
+          columns: processedData.columns,
+          keys: {
+            x: 'x'
+          }
+        };
+        widget.formattedData = data;
+
+        widget.pollId = res.__pollId__;
```

Comment:
It would be better if we have the pollId before the response (which I believe we have the ```dataSrc``` gets a promise and a __pollId__ so if we want to stop the polling even before we get the first response we could do it.